SignalServiceKit/src/Messages/Interactions/TSQuotedMessage.h

```diff
@@ -2,11 +2,47 @@
 //  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
 //
 
+#import <Mantle/MTLModel.h>
 #import <SignalServiceKit/TSYapDatabaseObject.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
-@interface TSQuotedMessage : TSYapDatabaseObject
+@class OWSSignalServiceProtosDataMessage;
+@class OWSSignalServiceProtosEnvelope;
+@class TSAttachment;
+@class TSAttachmentStream;
+@class TSQuotedMessage;
+@class TSThread;
+@class YapDatabaseReadWriteTransaction;
+
+@interface OWSAttachmentInfo: MTLModel
+
+@property (nonatomic, readonly, nullable) NSString *contentType;
+@property (nonatomic, readonly, nullable) NSString *sourceFilename;
+
+// This is only set when sending a new attachment so we have a way
+// to reference the original attachment when generating a thumbnail.
+// We don't want to do this until the message is saved, when the user sends
+// the message so as not to end up with an orphaned file.
+@property (nonatomic, readonly, nullable) NSString *attachmentId;
```

Comment:
So when we send an outgoing quoted reply, we have an instance of `OWSQuotedReplyModel` used in the authoring UI that generates a thumbnail, and then when we send the message, it is saved in the database and a new instance of `OWSQuotedReplyModel` (which makes its own thumbnail) is created by the conversation view (i.e. in the `ConversationViewItem`)?

Reply:
That's right. 

OWSQuotedReplyModel is not persisted, but serves as a "fully loaded" view model on top of the persisted TSQuotedMessage.